incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -218,6 +218,14 @@ func (r *ObjectReconciler) syncPropagated(ctx context.Context, log logr.Logger,
 		return remove, nil
 	}
 
+	// If an object doesn't exist, assume it's been deleted or not yet created. In this case, we will
+	// return write action and the source instance.
+	exist := inst.GetCreationTimestamp() != v1.Time{}
+	if !exist {
+		metadata.SetLabel(inst, api.LabelInheritedFrom, srcInst.GetNamespace())
```

Comment:
In order to avoid calling SetLabel twice, maybe merge the two `if` statements?

Reply:
Thanks for the suggestion! Done.